src/main/java/org/primefaces/extensions/validate/PhoneValidator.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2011-2019 PrimeFaces Extensions
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.primefaces.extensions.validate;
+
+import com.google.i18n.phonenumbers.NumberParseException;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import com.google.i18n.phonenumbers.Phonenumber;
+import javax.faces.application.FacesMessage;
+import javax.faces.component.UIComponent;
+import javax.faces.context.FacesContext;
+import javax.faces.validator.FacesValidator;
+import javax.faces.validator.Validator;
+import javax.faces.validator.ValidatorException;
+import org.primefaces.extensions.component.inputphone.InputPhone;
+
+/**
+ * Validator used with {@link InputPhone}.
+ *
+ * @author Jasper de Vries &lt;jepsar@gmail.com&gt;
+ */
+@FacesValidator("org.primefaces.extensions.validate.PhoneValidator")
+public class PhoneValidator implements Validator {
+
+    @Override
+    public void validate(FacesContext context, UIComponent component, Object object) throws ValidatorException {
+        String country = "";
+        if (InputPhone.class.isInstance(component)) {
+            InputPhone inputPhone = (InputPhone) component;
+            country = context.getExternalContext().getRequestParameterMap().get(inputPhone.getClientId() + "_iso2");
+            if (country == null || "auto".equals(country)) {
+                country = "";
+            }
+        }
+        try {
+            PhoneNumberUtil phoneNumberUtil = PhoneNumberUtil.getInstance();
+            Phonenumber.PhoneNumber phoneNumber = phoneNumberUtil.parse((String) object, country);
+            if (!phoneNumberUtil.isValidNumber(phoneNumber)) {
+                throw new ValidatorException(new FacesMessage("Invalid phone number"));
```

Comment:
I could not find strings that are loaded from a resource bundle in PF or PFE, so I hardcoded it for now. Is that okay? If so, I'll create a final static String as it is used twice.

Reply:
At least in PF there is a resource bundle _Messages_, see https://github.com/primefaces/primefaces/blob/96d30807fc67a2b6183f4956cb29202ae44759ee/src/main/java/org/primefaces/component/calendar/Calendar.java#L153 as an usage example.

Reply:
Thanks, I'll have a look at that.

Reply:
@melloware Would that mean that we need to introduce something like `PFEMessages`?

Reply:
Yep fee free to introduce a PFE messages like PF does it.  I will then look through all the PFE code and convert things to use that messages.

Reply:
Or if you want leave it hard coded for now, we can make it a clean up task for 7.1?

Reply:
@melloware Sounds like a bigger issue than this component alone. Let's create a new issue for this.